scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -87,6 +87,33 @@ impl ReplicatorConsumer {
         self.update_or_insert(data, true).await;
     }
 
+    // Recreates INSERT/DELETE statement on non-frozen map.
+    async fn update_map<'a>(
+        &self,
+        column_name: &str,
+        data: &'a CDCRow<'_>,
+        values_for_update: &mut Vec<&'a CqlValue>,
+        values_for_delete: &[&CqlValue],
+        timestamp: i64,
+    ) {
+        let deleted_elements: &'a [CqlValue] = data.get_deleted_elements(column_name);
+
+        if (data.get_value(column_name).is_none() && deleted_elements.is_empty())
+            || data.is_value_deleted(column_name)
```

Comment:
Please explain the reasoning behind the condition in a comment.

Reply:
It seems to me that the second part of the alternative is sufficient. INSERT and DELETE both overwrite the collection, inserting a collection-wide tombstone, so `is_value_deleted` is true.

Reply:
Yeah, you are right.